doc/source/dev/contributor/meson_advanced.rst

```diff
@@ -0,0 +1,52 @@
+.. _meson-advanced:
+
+Build with Meson Optimizations
+==============================
+
+Meson provides different optimizations levels while configuring the project. You can see
+the available options for optimizations at
+`meson documentation <https://mesonbuild.com/Builtin-options.html#core-options>`.
+
+Assuming that you are building from scratch(do ``git clean -xdf`` if needed), you can
+configure the build as following::
+
+    meson setup build --optimization g --prefix=$PWD/build-install
```

Comment:
It'd help to spell that `--optimization g` translates into `-Og` gcc switch (or does it?)

Reply:
Then, I expected that a page on meson optimizations would explain how does one specify further optimization switches,   `-march=native`, `-fopenmp` etc, is this planned?

Reply:
I have used the available options for optimizations from meson docs: https://mesonbuild.com/Builtin-options.html#core-options.

> that `--optimization g` translates into `-Og` gcc switch (or does it?)

Not sure about this.

Reply:
It does translate to `-Og` I believe. However, rather than `--optimization`, it may be better to use `buildtype` as the example I believe, see the second table under the link above. Those have plain names, like "plain, debug, debugoptimized, release, minsize".

Reply:
> However, rather than `--optimization`, it may be better to use `buildtype` as the example I believe,

Should I add a new section for `buildtype` or else just update the current section after removing `--optimization`?

Reply:
Updating the current section is what I had in mind.